src/System.Data.SqlClient/tests/FunctionalTests/SqlConnectionBasicTests.cs

```diff
@@ -135,6 +135,29 @@ public void ConnectionTimeoutTestWithThread()
             Console.WriteLine($"ConnectionTimeoutTestWithThread: Elapsed Time {theMax} and threshold {threshold}");
         }
 
+        [Fact]
+        public void ExceptionsWithMinPoolSizeCanBeHandled()
+        {
+            int count = 0;
+            int expected = 3;
+            string connectionString = $"Data Source={Guid.NewGuid().ToString()};uid=random;pwd=asd;Connect Timeout=2; Min Pool Size=3";
+            while (count < expected)
+            {
+                using (SqlConnection connection = new SqlConnection(connectionString))
+                {
+                    try
+                    {
+                        connection.Open();
+                    }
+                    catch (Exception)
+                    {
+                        count++;
+                    }
+                }
+            }
+            Assert.Equal(expected, count);
```

Comment:
If you do end up keeping the current structure, this Assert.Equal seems unnecessary... the only way you'll get here, by construction, is if expected == count, as that's when you exit the loop.